src/devices/cpu/drcbec.cpp

```diff
@@ -2232,8 +2231,7 @@ void drcbe_c::output_parameter(drcbec_instruction **dstptr, void **immedptr, int
 
 		// convert mapvars to immediates
 		case parameter::PTYPE_MAPVAR:
-			temp_param = m_map.get_last_value(param.mapvar());
-			return output_parameter(dstptr, immedptr, size, temp_param);
+			return output_parameter(dstptr, immedptr, size, param.mapvar());
```

Comment:
drcuml.cpp already has an optimization step which replaces mapvar references with constants when possible, except if the opcode is OP_RECOVER which requires the mapvar register ID. The other backends rely on that to get mapvar values. 

The C backend had an extra step where it'd try to get mapvar values during generation which would result in OP_RECOVER breaking because instead of getting the mapvar register ID, it'd get some other value. For example, if `uml::M1` was set to 1234 and OP_RECOVER was called for `uml::M1`, it'd try looking up the value from `uml::M0 + 1234` instead of `uml::M0 + 1`) if `uml::M1` was set within the same block, or if you're in a sub call somewhere it would fail to get the value of `uml::M1` and return 0 so it'd try to get the value from `uml::M0 + 0` which would also give a different value than you want.

https://github.com/mamedev/mame/blob/master/src/devices/cpu/drcuml.cpp#L421-L429